src/chat/participant.ts

```diff
@@ -213,14 +222,21 @@ export async function handleChatMessage(
         // each registered tool should contribute its own way of handling the invocation and
         // interacting with the stream, and can return an array of messages to be sent for another
         // round of processing
+
         logger.debug(`Processing tool invocation for "${toolCall.name}"`);
-        const newMessages: LanguageModelChatMessage[] = await tool.processInvocation(
-          request,
-          stream,
-          toolCall,
-          token,
-        );
-        toolResultMessages.push(...newMessages);
+        try {
+          const newMessages: LanguageModelChatMessage[] = await tool.processInvocation(
+            request,
+            stream,
+            toolCall,
+            token,
+          );
+          toolResultMessages.push(...newMessages);
+        } catch (error) {
+          logger.error(`Error processing tool invocation for "${toolCall.name}":`, error);
+          stream.markdown(`Error processing tool invocation for "${toolCall.name}": ${error}`);
+          continue;
```

Comment:
```suggestion
          logger.error(`Error processing tool invocation for "${toolCall.name}":`, error);
          messages.push(
            systemMessage(`Error processing tool invocation for "${toolCall.name}": ${error}`)
          );
          continue;
```
What do you think about pushing an error message to the model here? I think if we're going to stream the error markdown, we should probably exit this function instead of continuing the iteration.